core/src/test/java/cz/xtf/core/image/ImageTest.java

```diff
@@ -0,0 +1,59 @@
+package cz.xtf.core.image;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import cz.xtf.core.config.XTFConfig;
+import uk.org.webcompere.systemstubs.jupiter.SystemStub;
+import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension;
+import uk.org.webcompere.systemstubs.properties.SystemProperties;
+
+@ExtendWith(SystemStubsExtension.class)
+public class ImageTest {
+
+    public static final String EAP_IMAGE_URL = "registry.redhat.io/jboss-eap-8/eap8-openjdk17-builder-openshift-rhel8:latest";
+    public static final String EAP_IMAGE_SUBID_NAME = "wildfly-openjdk17";
+    public static final String EAP_IMAGE_SUBID_URL = "quay.io/wildfly/wildfly-s2i:latest-jdk17";
+
+    @SystemStub
+    private SystemProperties systemProperties;
+
+    @Test
+    public void testResolveImageWithSubId() {
+        System.setProperty("xtf.eap.image", EAP_IMAGE_URL);
```

Comment:
use SystemStubs `SystemProperties systemProperties` 

```
systemProperties.set("xtf.eap.image", EAP_IMAGE_URL);
```

Reply:
IIUC this approach is ok: https://github.com/webcompere/system-stubs/blob/main/system-stubs-jupiter/README.md#by-field

Reply:
```
    // this test cannot damage the system properties
    // because changes made via System.setProperty
    // or systemProperties.set will be undone after the test
```
ah ok, anyway I would prefer using `SystemProperties` otherwise it is not clear why that field is injected

Reply:
Anyway, I’ve made the change. :-)